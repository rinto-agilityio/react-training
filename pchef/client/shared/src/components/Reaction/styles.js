// Libs
import { StyleSheet } from 'react-native'

// Themes
import { METRICS } from '../../themes'

const styles = StyleSheet.create({
  wrapperIcon: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    padding: METRICS.padding.lg,
  },
})

export default styles
