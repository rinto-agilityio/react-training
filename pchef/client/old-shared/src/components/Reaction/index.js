// Libs
import React from 'react'
import { View } from 'react-native'

// Styles
import styles from './styles'

// Themes
import { COLORS, METRICS } from '../../themes'

// Components
import Icon from '../Icon'

type Props = {
  isFavorited: boolean,
  customWrapperIcon?: {},
  size: string,
  votes: Array<number>,
  onClickFavorite?: () => void,
  onClickVote?: () => void,
}

const Reaction = ({
  isFavorited,
  customWrapperIcon,
  onClickFavorite,
  onClickVote,
  size = '',
  votes = [],
}: Props) => {
  // Data render reaction icon
  const data = [
    {
      name: isFavorited ? 'heart' : 'heart-o',
      color: COLORS.grayDarker,
      label: 'Save',
      onClick: onClickFavorite,
    },
    {
      name: 'thumbs-up',
      color: COLORS.grayDarker,
      label: votes.length,
      onClick: onClickVote,
    },
  ]

  return (
    <View style={[styles.wrapperIcon, customWrapperIcon]}>
      {data.map(item => (
        <Icon
          key={item.name}
          name={item.name}
          size={METRICS[`${size}Icon`]}
          color={item.color}
          onClick={item.onClick}
          underlayColor="transparent"
          label={item.label}
        />
      ))
      }
    </View>
  )
}

Reaction.defaultProps = {
  customWrapperIcon: {},
  onClickFavorite: () => {},
  onClickVote: () => {},
}

export default Reaction
