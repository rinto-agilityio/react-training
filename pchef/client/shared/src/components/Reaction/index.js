// Libs
import React from 'react'
import { View } from 'react-native'

// Styles
import styles from './styles'

// Themes
import { COLORS, METRICS } from '../../themes'

// Components
import Icon from '../Icon'

type Props = {
  isFavorited?: boolean,
  isVote?: boolean,
  customWrapperIcon?: {},
  size: string,
  votes: Array<string>,
  onPressFavorite?: () => void | Promise<void>,
  onPressVote?: () => void | Promise<void>,
}

const Reaction = ({
  isFavorited,
  customWrapperIcon,
  onPressFavorite,
  onPressVote,
  isVote,
  size = '',
  votes = [],
}: Props) => {
  // Data render reaction icon
  const data = [
    {
      name: 'favorite',
      color: isFavorited ? COLORS.blue : COLORS.grayDarker,
      label: 'Save',
      onPress: onPressFavorite,
    },
    {
      name: 'thumb-up',
      color: isVote ? COLORS.blue : COLORS.grayDarker,
      label: votes.length,
      onPress: onPressVote,
    },
  ]

  return (
    <View style={[styles.wrapperIcon, customWrapperIcon]}>
      {data.map(item => (
        <Icon
          key={item.name}
          name={item.name}
          size={METRICS[`${size}Icon`]}
          color={item.color}
          onPress={item.onPress}
          underlayColor="transparent"
          label={item.label}
        />
      ))}
    </View>
  )
}

Reaction.defaultProps = {
  customWrapperIcon: {},
  onPressFavorite: () => {},
  onPressVote: () => {},
  isFavorited: false,
  isVote: false,
}

export default Reaction
