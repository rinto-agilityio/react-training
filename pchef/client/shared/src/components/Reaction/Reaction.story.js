// Libs
import React from 'react'
import { View } from 'react-native'
import { storiesOf } from '@storybook/react'

// Themes
import { METRICS } from '../../themes'

// Components
import Reaction from '.'

storiesOf('Reaction', module)
  .add('favorited', () => (
    <View
      style={{
        width: METRICS.largeBgImage,
        height: METRICS.largeBgImage,
      }}
    >
      <Reaction
        votes={[1]}
        size="medium"
        isFavorited
      />
    </View>
  ))
  .add('not favorited', () => (
    <View
      style={{
        width: METRICS.largeBgImage,
        height: METRICS.largeBgImage,
      }}
    >
      <Reaction
        votes={[1]}
        size="medium"
      />
    </View>
  ))
