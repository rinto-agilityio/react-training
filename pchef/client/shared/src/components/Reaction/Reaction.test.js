// Components
import Reaction from '.'

it('renders correctly', () => {
  const reaction = renderer.create(
    <Reaction
      votes={[1]}
      size="medium"
      isFavorited
    />,
  ).toJSON()

  expect(reaction).toMatchSnapshot()
})
