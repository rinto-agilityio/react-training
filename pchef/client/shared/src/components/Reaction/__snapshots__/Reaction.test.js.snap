// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-view-1dbjc4n r-display-6koalj r-flexDirection-18u37iz r-justifyContent-1wtj0ep r-padding-1pcd2l5"
>
  <div
    className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz"
  >
    <div
      className="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1niwhzg r-borderRadius-6ncur5 r-cursor-1loqt21 r-height-1r8g8re r-justifyContent-1777fci r-margin-5scogr r-overflow-bztko3 r-touchAction-1otgn73 r-userSelect-lrvibr r-width-1acpoxo"
      data-focusable={true}
      disabled={false}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="css-hitSlop-mjp8i1"
        style={
          Object {
            "bottom": "-6px",
            "left": "-6px",
            "right": "-6px",
            "top": "-6px",
          }
        }
      />
      <div
        className="css-view-1dbjc4n"
      >
        <div
          className="css-text-76zvg2 r-backgroundColor-1niwhzg r-pointerEvents-633pao"
          dir="auto"
          focusable={false}
          role="img"
          style={
            Object {
              "color": "rgba(128,128,128,1.00)",
              "fontSize": "30px",
            }
          }
        >
          □
        </div>
      </div>
    </div>
    <div
      className="css-text-76zvg2 r-marginLeft-1n0xq6e"
      dir="auto"
    >
      Save
    </div>
  </div>
  <div
    className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz"
  >
    <div
      className="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1niwhzg r-borderRadius-6ncur5 r-cursor-1loqt21 r-height-1r8g8re r-justifyContent-1777fci r-margin-5scogr r-overflow-bztko3 r-touchAction-1otgn73 r-userSelect-lrvibr r-width-1acpoxo"
      data-focusable={true}
      disabled={false}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="css-hitSlop-mjp8i1"
        style={
          Object {
            "bottom": "-6px",
            "left": "-6px",
            "right": "-6px",
            "top": "-6px",
          }
        }
      />
      <div
        className="css-view-1dbjc4n"
      >
        <div
          className="css-text-76zvg2 r-backgroundColor-1niwhzg r-pointerEvents-633pao"
          dir="auto"
          focusable={false}
          role="img"
          style={
            Object {
              "color": "rgba(128,128,128,1.00)",
              "fontSize": "30px",
            }
          }
        >
          □
        </div>
      </div>
    </div>
    <div
      className="css-text-76zvg2 r-marginLeft-1n0xq6e"
      dir="auto"
    >
      1
    </div>
  </div>
</div>
`;
